crates/storage/src/db/mod.rs

```diff
@@ -0,0 +1,133 @@
+pub mod beacon;
+pub mod lean;
+
+use std::{fs, io, path::PathBuf, sync::Arc};
+
+use anyhow::Result;
+use beacon::BeaconDB;
+use lean::LeanDB;
+use redb::{Builder, Database};
+use tracing::info;
+
+use crate::{
+    errors::StoreError,
+    tables::{
+        beacon::{
+            beacon_block::BEACON_BLOCK_TABLE, beacon_state::BEACON_STATE_TABLE,
+            blobs_and_proofs::BLOB_FOLDER_NAME, block_timeliness::BLOCK_TIMELINESS_TABLE,
+            checkpoint_states::CHECKPOINT_STATES_TABLE,
+            equivocating_indices::EQUIVOCATING_INDICES_FIELD,
+            finalized_checkpoint::FINALIZED_CHECKPOINT_FIELD, genesis_time::GENESIS_TIME_FIELD,
+            justified_checkpoint::JUSTIFIED_CHECKPOINT_FIELD,
+            latest_messages::LATEST_MESSAGES_TABLE,
+            parent_root_index::PARENT_ROOT_INDEX_MULTIMAP_TABLE,
+            proposer_boost_root::PROPOSER_BOOST_ROOT_FIELD, slot_index::SLOT_INDEX_TABLE,
+            state_root_index::STATE_ROOT_INDEX_TABLE, time::TIME_FIELD,
+            unrealized_finalized_checkpoint::UNREALIZED_FINALIZED_CHECKPOINT_FIELD,
+            unrealized_justifications::UNREALIZED_JUSTIFICATIONS_TABLE,
+            unrealized_justified_checkpoint::UNREALIZED_JUSTIFED_CHECKPOINT_FIELD,
+        },
+        lean::{
+            lean_block::LEAN_BLOCK_TABLE, lean_state::LEAN_STATE_TABLE,
+            slot_index::LEAN_SLOT_INDEX_TABLE, state_root_index::LEAN_STATE_ROOT_INDEX_TABLE,
+        },
+    },
+};
+
+pub const REDB_BEACON_FILE: &str = "ream.redb";
+pub const REDB_LEAN_FILE: &str = "lean.redb";
+
+/// The size of the cache for the database
+///
+/// 1 GiB
+pub const REDB_CACHE_SIZE: usize = 1_024 * 1_024 * 1_024;
+
+#[derive(Clone, Debug)]
+pub struct ReamDB {
+    db: Arc<Database>,
+}
+
+impl ReamDB {
+    pub fn new(data_dir: PathBuf) -> Result<Self, StoreError> {
+        let db = Builder::new()
+            .set_cache_size(REDB_CACHE_SIZE)
+            .create(data_dir.join(REDB_BEACON_FILE))?;
```

Comment:
```suggestion
impl ReamDB {
    pub fn new(data_dir: PathBuf) -> Result<Self, StoreError> {
        let db = Builder::new()
            .set_cache_size(REDB_CACHE_SIZE)
            .create(data_dir.join(REDB_FILE))?;
```

We should only use one database file, we could be running lean,beacon, or lean+beacon for example during the transition